File: Clients/BlogBlazor/Shared/Blazor.Contracts/V1/Validators/TagsValidators/TagRequestValidator.cs
using Blazor.Contracts.V1.Requests.TagsRequests;
using FluentValidation;

namespace Blazor.Contracts.V1.Validators.TagsValidators
{
    /// <summary>
    /// Tag request validator.
    /// </summary>
    /// <seealso cref="AbstractValidator{TagRequest}" />
    public class TagRequestValidator : AbstractValidator<TagRequest>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="TagRequestValidator"/> class.
        /// </summary>
        public TagRequestValidator()
        {
            RuleFor(x => x.Title)
                .NotEmpty()
                .MinimumLength(2)
                .MinimumLength(64)
                .Matches("^[a-zA-Z0-9 ]*S");
        }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/TagsValidators/TagRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.TagsValidators

open FluentValidation
open Blog.FSharp.Contracts.V1.Requests.TagsRequests

/// <summary>
/// Validator for a tag request.
/// </summary>
type TagRequestValidator() =
    inherit AbstractValidator<TagRequest>()

    do
        // Ensure Title is not empty, has a length between 2 and 64, and matches the regex
        base.RuleFor(fun x -> x.Title)
            .NotEmpty()
            .MinimumLength(2)
            .MaximumLength(64)
            .Matches("^[a-zA-Z0-9 ]*S") |> ignore